### PR TITLE
Refactor kubernetes initialization

### PR DESCRIPTION
**Pull request by relu**
Using the same logic implemented in the kubernetes provider

Related to https://github.com/terraform-providers/terraform-provider-helm/pull/378#issuecomment-573230759

### CODE REVIEW

**Comment by aaronmell** [comment]
Change seems fine to me, could use a test.

**Comment by relu** [comment]
The implementation is pretty much a copy-paste from the kubernetes provider and bits taken from helm itself, looks a bit more solid than the previous tempfile based approach and would also introduce support for exec auth (eks would benefit from this).
The only problem is that the namespace is not being properly set for releases causing all resources to be created under the `default` namespace. Maybe you have some idea about how we can tackle this issue? I wasn't able to figure out a solution I'm afraid.

File: helm/provider.go [hunk 1]
```diff
@@ -377,33 +484,14 @@ func (m *Meta) GetHelmConfiguration(namespace string) (*action.Configuration, er
 	defer m.Unlock()
 
 	actionConfig := new(action.Configuration)
-	cf := getKubernetesConfiguration(m.KubernetesConfig, namespace)
 
-	if err := actionConfig.Init(cf, namespace, m.HelmDriver, debug); err != nil {
+	if err := actionConfig.Init(m.KubeConfig, namespace, m.HelmDriver, debug); err != nil {
```
Comment (aaronmell) [inline helm/provider.go]: I think you need to modify m.Kubeconfig and set the namespace on that object as well.

**Comment by relu** [comment]
I managed to get it working. The solution was pretty simple: initialize one client per action rather than one client globally during the provider execution runtime. Tests ran successfully on my local machine.

**Pull request merged**